sretoolbox/container/image.py

```diff
@@ -353,9 +382,17 @@ def _request_get(self, url):
 
             # Try again, this time with the Authorization header
             headers['Authorization'] = self._get_auth(www_auth)
-            response = requests.get(url, headers=headers)
+            response = requests.head(url, headers=headers)
```

Comment:
Before making this request, It's cheaper to first check if we have a cache record for that url.

Reply:
Not really, because the object an URL points to can change. For instance, when operating with a `latest` image, that tag can move. My cache key is the etag - that's what we've been discussing all around - not the URL.

Reply:
If there's nothing in the cache, no need for HEAD requests.

Reply:
I disagree - as of today, the API for the cache states the key is the `etag` - this is what was agreed on the design document. Therefore there is no better-than-O(N) way to look up a given URL. This is also what prompted the separation of the manifest-specific caching, let's not move the goal posts.

Then, this is also assuming that it isn't possible for objects to get renamed, that there will never be any redirects or that different mirrors won't hold the same objects. I doubt those assumptions belong in a simple wrapper around HTTP `GET`.

If we decided etags rule (and for good reason), then we have to fetch the etag first.

Reply:
We are mixing a bit the topics here. But it's fine, we can always improve, if/when performance is an issue. No need to improve for performance beforehand, if that's not in line with the design you believe in.
Thanks for taking it into consideration anyway.